### PR TITLE
PR Allure

### PR DESCRIPTION
**Pull request by raulgalera97**
All changes are donde in Allure branch, and is ready to merge with main now.

### CODE REVIEW

File: src/test/java/utilities/DriverConfiguration.java [hunk 1]
```diff
@@ -7,37 +7,69 @@
...
-    } else return null;
+    } else {
+      return null;
+    }
+  }
+
+  private WebDriver configureWebDriver() {
+    WebDriver driver;
+    String browser = LocalEnviroment.getBrowser();
+    if (browser == null) {
```
Comment (raunelgarcia) [inline src/test/java/utilities/DriverConfiguration.java]: Objects.isNull(browser)

File: src/test/java/utilities/DriverConfiguration.java [hunk 2]
```diff
@@ -7,37 +7,69 @@
...
+      browser = "chrome";
+    } else {
+      browser = browser.toLowerCase();
```
Comment (raunelgarcia) [inline src/test/java/utilities/DriverConfiguration.java:49]: remove unnecesary else. Put directly when browser variable is declared as String browser = LocalEnviroment.getBrowser().toLowerCase

File: src/test/java/utilities/LocalEnviroment.java [hunk 3]
```diff
@@ -27,14 +28,21 @@ public static String getAppActivity() {
...
+
+  public static String getResolution() {
+    return System.getenv("Resolution");
   }
 
   public static boolean isMobile() {
     String platform = System.getenv("Platform");
-    if (Objects.equals(platform, "Android")) return true;
-    else return false;
+    return Objects.nonNull(platform) && platform.equalsIgnoreCase("Android");
```
Comment (raunelgarcia) [inline src/test/java/utilities/LocalEnviroment.java]: || platform.equalsIgnoreCase("IOS")

File: src/test/java/utilities/ScreenResolution.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  private static boolean isValidResolution(String resolution, String browser) throws IllegalStateException {
+    Map<String, Map<String, Map<String, List<String>>>> allowableResolutions = loadAllowableResolutionsFromYamlFile();
+
+    if (Objects.isNull(allowableResolutions))
+      throw new IllegalStateException("The \"allowable-resolutions.yaml\" file could not be located or loaded");
+
+    return allowableResolutions.get("browsers").get(browser).get("resolutions").contains(resolution);
+  }
+
+  public static Map<String, Map<String, Map<String, List<String>>>> loadAllowableResolutionsFromYamlFile() {
```
Comment (raunelgarcia) [inline src/test/java/utilities/ScreenResolution.java]: rename to loadAllowedResolutions

File: README.md [hunk 5]
```diff
@@ -52,10 +52,23 @@ Navigate to the directory where you cloned the repository and select it.
 You can set environment variables in IntelliJ IDEA by going to Run > Edit Configurations > Add JUNIT Configuration > Environment.
 
 ### Android Variables
-AppActivity=xxx;AppPackage=xxxx;Platform=Android;Udid=xxxx
```
Comment (raunelgarcia) [inline README.md:55]: add this environment variables as it was too

File: src/test/java/utilities/DriverConfiguration.java (hunk 2 above)
Comment (raulgalera97) [reply src/test/java/utilities/DriverConfiguration.java:49]: cant change this logic because if browser is null you cant call method "toLowerCase()"

**Pull request merged**